torch/_torch_docs.py

```diff
@@ -8464,10 +8464,17 @@ def merge_dicts(*dicts):
 Also supports batched inputs, and, if the input is batched, the output is batched with the same dimensions.
 
 .. seealso::
-
         :func:`torch.geqrf` can be used to form the Householder representation `(input, tau)` of matrix `Q`
         from the QR decomposition.
 
+.. note::
+        This function supports backward. However, unless ``(input, tau)`` do not require gradients and/or
+        ``tau.size(-1)`` is very small, it is recommended to use a combination of :func:`torch.linalg.qr`
+        with :func:`torch.matmul` for much faster backward. Note that the backward for :func:`torch.linalg.qr`
+        is only stable for full-rank inputs, and if the input is not full-rank, it is better to use
+        :func:`torch.ormqr` trading performance for numerical stability.
```

Comment:
The goal it to propagate through basis which `a, tau` encode.

Reply:
My point is: The QR decomposition is not differentiable on matrices that are not full-rank. That is, `geqrf` and `linalg.qr` are not differentiable on these inputs. The outputs of those functions are the same only that encoded different (there is a diffeomorphism between their outputs, the diffeomorphism being `linalg.householder_product`). Then, the `ormqr` and `matmul` are equivalent, only that `ormqr` works on one of the spaces, and `matmul` works in the other. As such, this function is as good as `matmul`. This is why comparing this against `linalg.qr` does not make sense. It just makes senes to compare `geqrf` with `linalg.qr`, and neither of them is differentiable on non-full-rank matrices.

Reply:
Sure, but otherwise it still simplifies to a transformation of the form of the `householder_product` and a matmul with a few matrices, so the reasoning above still applies in this sense. My point is that this operation has intrinsically the same properties as `matmul` for this reason.

Reply:
So, what are you proposing to add into the doc?